StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetConfiguration.swift

```diff
@@ -307,4 +328,52 @@ extension PaymentSheet {
         public var phone: String?
     }
 
+    /// 🚧 Under construction
+    /// Configuration for how billing details are collected during checkout.
+    @_spi(STP) public struct BillingDetailsCollectionConfiguration: Equatable {
+        /// Billing details fields collection options.
+        public enum CollectionMode: CaseIterable {
+            /// The field will be collected depending on the Payment Method's requirements.
+            case automatic
+            /// The field will never be collected.
+            /// If this field is required by the Payment Method, you must provide it as part of `defaultBillingDetails`.
+            case never
+            /// The field will always be collected, even if it isn't required for the Payment Method.
+            case always
+        }
+
+        /// Billing address collection options.
+        public enum AddressCollectionMode: CaseIterable {
+            /// Only the fields required by the Payment Method will be collected, this may be none.
+            case automatic
+            /// Address will never be collected.
+            /// If the Payment Method requires a billing address, you must provide it as part of
+            /// `defaultBillingDetails`.
+            case never
+            /// Collect the full billing address, regardless of the Payment Method requirements.
+            case full
+        }
+
+        /// How to collect the name field.
+        /// Defaults to `automatic`.
+        public var name: CollectionMode = .automatic
+
+        /// How to collect the phone field.
+        /// Defaults to `automatic`.
+        public var phone: CollectionMode = .automatic
+
+        /// How to collect the email field.
+        /// Defaults to `automatic`.
+        public var email: CollectionMode = .automatic
+
+        /// How to collect the billing address.
+        /// Defaults to `automatic`.
+        public var address: AddressCollectionMode = .automatic
+
+        /// Whether the values included in `Configuration.defaultBillingDetails` should be attached to the payment
+        /// method, this includes fields that aren't displayed in the form.
+        ///
+        /// If `false` (the default), those values will only be used to prefill the corresponding fields in the form.
+        public var attachDefaultsToPaymentMethod = false
```

Comment:
I added this field as a temporary way to not attach the defaults to avoid breaking the current functionality while this is in development, but the more I think about it, the more I think it is a good idea to add this in the final product so people have to opt-into this behavior, if people think this is a good idea I'll add it to the API review.

Reply:
After discussion, this field will be part of the final public API